rust/gui-client/src-tauri/src/client/gui/system_tray.rs

```diff
@@ -33,15 +36,75 @@ pub(crate) struct Tray {
     last_icon_set: Icon,
 }
 
-pub(crate) enum Menu<'a> {
+pub(crate) enum AppState<'a> {
     Loading,
     SignedOut,
     WaitingForBrowser,
     WaitingForConnlib,
-    SignedIn {
-        actor_name: &'a str,
-        resources: &'a [ResourceDescription],
-    },
+    SignedIn(SignedIn<'a>),
+}
+
+pub(crate) struct SignedIn<'a> {
+    pub(crate) actor_name: &'a str,
+    pub(crate) favorite_resources: &'a HashSet<ResourceId>,
+    pub(crate) resources: &'a [ResourceDescription],
+}
+
+impl<'a> SignedIn<'a> {
+    fn is_favorite(&self, res: &ResourceDescription) -> bool {
+        self.favorite_resources.contains(&res.id())
+    }
+
+    /// Builds the submenu that has the resource address, name, desc,
+    /// sites online, etc.
+    fn resource_submenu(&self, res: &ResourceDescription) -> Menu {
+        let submenu = Menu::default().add_item(resource_header(res));
+
+        let submenu = if self.is_favorite(res) {
+            submenu.add_item(item(Event::RemoveFavorite(res.id()), "Remove favorite").selected())
+        } else {
+            submenu.item(Event::AddFavorite(res.id()), "Add favorite")
+        };
+
+        let submenu = submenu
+            .separator()
+            .disabled("Resource")
+            .copyable(res.name())
+            .copyable(res.pastable().as_ref());
+
+        if let Some(site) = res.sites().first() {
+            // Emojis may be causing an issue on some Ubuntu desktop environments.
+            let status = match res.status() {
+                Status::Unknown => "[-] No activity",
+                Status::Online => "[O] Gateway connected",
+                Status::Offline => "[X] All Gateways offline",
+            };
+
+            submenu
+                .separator()
+                .disabled("Site")
+                .copyable(&site.name) // Hope this is okay - The code is simpler if every enabled item sends an `Event` on click
```

Comment:
I can change this back to the old behavior where we have some rare items that are enabled but clicking does nothing, by making `enabled` explicit. I just thought it's weird to have a button that does nothing, at least if it copies the text maybe it'll aid debugging or something.